src/neuroconv/tools/nwb_helpers/_models/_base_models.py

```diff
@@ -61,6 +103,22 @@ def __init__(self, **values):
         values.update(dataset_name=dataset_name)
         super().__init__(**values)
 
+    @classmethod
+    def from_neurodata_object(cls, neurodata_object: Container, field_name: str) -> "DatasetInfo":
```

Comment:
```suggestion
    def from_neurodata_object(cls, neurodata_object: Container, field_name: str) -> Self:
```

Shouldn't this use [`typing.Self`](https://peps.python.org/pep-0673/#use-in-classmethod-signatures)?

Reply:
Does that work? I thought that it was introduced in python 3.11 as the PEP states.

Reply:
Right you are; will raise this as an issue to not forget then